src/use_debounced_callback/index.jsx

```diff
@@ -1,11 +1,18 @@
 import debounce from 'lodash/debounce';
-import { useMemo } from 'react';
-
-const useDebouncedCallback = (func, delay) => (
-  useMemo(
-    () => debounce(func, delay),
-    [func, delay],
-  )
-);
+import { useEffect, useState } from 'react';
+
+const useDebouncedCallback = (func, delay) => {
+  const [instance, setInstance] = useState(undefined);
```

Comment:
Possible issue: `useDebouncedCallback` will return `undefined` on first render because we use `useEffect`. Maybe we should switch over to using `useLayoutEffect` with some guard condition for SSR?

Reply:
```
// use-isomorphic-layout-effect.js
import { useLayoutEffect, useEffect } from 'react';
const useIsomorphicLayoutEffect =
  typeof window !== 'undefined' ? useLayoutEffect : useEffect;
export default useIsomorphicLayoutEffect;
```

Mb smt like this to be sure?